blueflood-elasticsearch/src/main/java/com/rackspacecloud/blueflood/io/ElasticsearchRestHelper.java

```diff
@@ -464,8 +472,10 @@ public int refreshIndex(String indexName) throws IOException {
         HttpGet httpGet = new HttpGet(url);
         CloseableHttpResponse response = null;
         try {
+            this.publishPoolStats();
             response = closeableHttpClient.execute(httpGet);
         } catch (IOException e) {
+            counter(getClass(), "refreshIndex Error").inc();
```

Comment:
Should this be assigned to a field like the other metrics here? That seems to be the convention in the project. Not sure if there's any performance overhead to obtaining it on demand like this.

Reply:
Not a significant hit to performance as the metric needs to be created/lookedup only for Exception scenarios. But it does make sense make it instance field. Fixed it.